src/components/tests/SelectedRoute.test.tsx

```diff
@@ -0,0 +1,184 @@
+const mockGetDottedLineStyle = uiService.getDottedLineStyle as jest.MockedFunction<typeof uiService.getDottedLineStyle>;
+const mockGetStationIconStyle = uiService.getStationIconStyle as jest.MockedFunction<typeof uiService.getStationIconStyle>;
+
+import * as utilService from "../../services/util.service";
+jest.mock("../../services/util.service");
+
+const mockGetStation = utilService.getStation as jest.MockedFunction<typeof utilService.getStation>;
+const mockGetStationName = utilService.getStationName as jest.MockedFunction<typeof utilService.getStationName>;
+
+let mockedTranslate = jest.fn();
+let mockedChangeLanguage = jest.fn();
+jest.mock("react-i18next", () => {
+  return {
+    useTranslation: () => {
+      return ({
+        i18n: {
+          changeLanguage: mockedChangeLanguage,
+        },
+        t: mockedTranslate,
+      })
+    }
+  }
+});
+
+describe("<SelectedRoute />", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mockGetStation.mockReturnValue(station);
+  });
+
+  it("should render the component without errors", async () => {
+    render(<SelectedRoute travelRoute={travelRoute}/>);
+  });
+
+  it("should show the fare and translated text", async () => {
+    mockedTranslate.mockReturnValue("Fare");
+    const travelRouteWith150Fare = {
+      ...travelRoute,
+      fare: 150
+    }
+    const { findByTestId } = render(<SelectedRoute travelRoute={travelRouteWith150Fare}/>);
+
+    const component =  await findByTestId("selected-route-fare");
+
+    expect(mockedTranslate.mock.calls[0][0]).toBe("route.fare");
+    expect(component).toHaveTextContent("Fare: 150");
+  });
+
+  it("should show the container with correct classname", async () => {
+    const { findByTestId } = render(<SelectedRoute travelRoute={travelRoute}/>);
+
+    const component =  await findByTestId("selected-route-container");
+
+    expect(component).toHaveClass("travel-route-container");
+  });
+
+  it("should call getInterChangeLine if there are more than 2 route segments", async () => {
+    const travelRouteWith3Routes = {
+      ...travelRoute,
+      route: [
+        { route: [], lineType: LineType.MRT_BLUE, fare: 0},
+        { route: [], lineType: LineType.BTS_SUKHUMVIT, fare: 0},
+        { route: [], lineType: LineType.BTS_SILOM, fare: 0},
+      ]
+    }
+    mockGetInterChangeLine.mockReturnValueOnce("arl-dotted-line").mockReturnValueOnce("interchange-dotted-line");
+    const { findAllByTestId } = render(<SelectedRoute travelRoute={travelRouteWith3Routes}/>);
+
+    const components =  await findAllByTestId("selected-route-interchange-dotted-line-style");
+
+    expect(components.length).toBe(3);
+    expect(components[1]).toHaveClass("arl-dotted-line");
+    expect(components[2]).toHaveClass("interchange-dotted-line");
+    expect(mockGetInterChangeLine.mock.calls[0]).toEqual([LineType.BTS_SUKHUMVIT, LineType.MRT_BLUE]);
```

Comment:
for this test, i think mock return value as `arl-dotted-line` is not reflect the given travelRoute input and might lead to confusion since it is not the expected behavior that it should does.
if you need to mock return value with many condition. you can try this:
```
mockGetInterChangeLine.mockImplementation((input) => {
    if (input === 'A') { return  something }
    return something
});
```